programs/vote/src/vote_state/mod.rs

```diff
@@ -1114,6 +1156,98 @@ mod tests {
+
+        // Create a fake TransactionContext with a fake InstructionContext with a single account which is the
+        // vote account that was just created
+        let transaction_context =
+            TransactionContext::new(vec![(node_pubkey, vote_account)], None, 0, 0);
+        let mut instruction_context = InstructionContext::default();
+        instruction_context.configure(
+            &[0],
+            &[InstructionAccount {
+                index_in_transaction: 0,
+                index_in_caller: 0,
+                index_in_callee: 0,
+                is_signer: false,
+                is_writable: true,
+            }],
+            &[],
+        );
+
+        // Get the BorrowedAccount from the InstructionContext which is what is used to manipulate and inspect account
+        // state
+        let mut borrowed_account = instruction_context
+            .try_borrow_instruction_account(&transaction_context, 0)
+            .unwrap();
+
+        // Ensure that the vote state started out at 1_14_11
+        let vote_state_version = borrowed_account.get_state::<VoteStateVersions>().unwrap();
+        assert!(matches!(vote_state_version, VoteStateVersions::V1_14_11(_)));
+
+        // Convert the vote state to current as would occur during vote instructions
+        let vote_state = vote_state_version.convert_to_current();
+
+        // Now re-set the vote account state; because the feature is not enabled, the old 1_14_11 format should be
+        // written out
+        assert_eq!(
+            set_vote_account_state(&mut borrowed_account, vote_state.clone(), &feature_set),
+            Ok(())
+        );
+        let vote_state_version = borrowed_account.get_state::<VoteStateVersions>().unwrap();
+        assert!(matches!(vote_state_version, VoteStateVersions::V1_14_11(_)));
+
+        // Test that when the feature is enabled, if the vote account does not have sufficient lamports to realloc,
+        // the old vote state is written out
+        feature_set.activate(&feature_set::vote_state_add_vote_latency::id(), 1);
+        assert_eq!(
+            set_vote_account_state(&mut borrowed_account, vote_state.clone(), &feature_set),
+            Ok(())
+        );
+        let vote_state_version = borrowed_account.get_state::<VoteStateVersions>().unwrap();
+        assert!(matches!(vote_state_version, VoteStateVersions::V1_14_11(_)));
+
+        // Test that when the feature is enabled, if the vote account does have sufficient lamports, the
+        // new vote state is written out
+        assert_eq!(
+            borrowed_account.set_lamports(rent.minimum_balance(VoteState::size_of())),
+            Ok(())
+        );
+        assert_eq!(
+            set_vote_account_state(&mut borrowed_account, vote_state, &feature_set),
+            Ok(())
+        );
+        let vote_state_version = borrowed_account.get_state::<VoteStateVersions>().unwrap();
+        assert!(matches!(vote_state_version, VoteStateVersions::Current(_)));
```

Comment:
Might be good to verify here that the fields are accurate and reflect the old state as well. For that purpose, it would be good to have old state not have default fields

Reply:
I will do this, but am traveling this upcoming week so this will be delayed until after that, thanks!

Reply:
Hi sorry for the delay, I have updated the test to include nonzero vote state and verify that the vote state has not changed throughout the upgrade tests.